fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/serializer/JsonCastVoteDeserializer.java

```diff
@@ -17,44 +16,45 @@
 
 public class JsonCastVoteDeserializer implements JsonDeserializer<CastVote> {
 
-    private static final String VOTE_INDEX = "vote";
-
-    @Override
-    public CastVote deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
-        JsonObject obj = json.getAsJsonObject();
-        JsonArray jsonVote = obj.getAsJsonArray("votes");
-
-        // Parse fields of the Json
-        JsonElement electionIdField = json.getAsJsonObject().get("election");
-        JsonElement laoIdField = json.getAsJsonObject().get("lao");
-        JsonElement createdAtField = json.getAsJsonObject().get("created_at");
-
-        // Get content
-        String electionId = context.deserialize(electionIdField, String.class);
-        String laoId = context.deserialize(laoIdField, String.class);
-        long createdAt = context.deserialize(createdAtField, long.class);
-
-        boolean typeValidationInt = true;
-        boolean typeValidationString = true;
-        // Vote type of a CastVote is either an integer for an OpenBallot election or a
-        // String for an Encrypted election, type should be valid for all votes
-        for (int i = 0; i < jsonVote.size(); i++) {
-            JsonObject voteContent = jsonVote.get(i).getAsJsonObject();
-            typeValidationInt =
-                    typeValidationInt && voteContent.get(VOTE_INDEX).getAsJsonPrimitive().isNumber();
-            typeValidationString =
-                    typeValidationString && voteContent.get(VOTE_INDEX).getAsJsonPrimitive().isString();
-        }
-        if (typeValidationInt && !typeValidationString) {
-            Type token = new TypeToken<List<ElectionVote>>() {}.getType();
-            List<ElectionVote> votes = context.deserialize(jsonVote, token);
-            return new CastVote(votes, electionId, laoId, createdAt);
-        } else if (!typeValidationInt && typeValidationString) {
-            Type token = new TypeToken<List<ElectionEncryptedVote>>() {}.getType();
-            List<ElectionEncryptedVote> votes = context.deserialize(jsonVote, token);
-            return new CastVote(votes, electionId, laoId, createdAt);
-        } else {
-            throw new JsonParseException("Unknown vote type in cast vote message");
-        }
+  private static final String VOTE_INDEX = "vote";
```

Comment:
This is strange, why did it get deleted ?

Reply:
Hey, If you take again a new look, it is not deleted, it is just Zoe editor that has reformated the whole file and github counts It (weirdly idk why) as deletion follow of the same thing that is added.